### DISCUSSION TITLE
Demo: Instream video ad in a Protected Audience sequential auction setup

### DISCUSSION THREAD

**Discussion by kevinkiklee** [privacysandbox/privacy-sandbox-demos#254] [General]
We have released a demo for an instream video ad in a Protected Audience sequential auction setup. 

If you have any questions or comments regarding the video demo, feel free to reply to this discussion post. 

* Documentation: https://privacy-sandbox-demos-home.dev/docs/demos/instream-video-ad-multi-seller 
* PR: https://github.com/privacysandbox/privacy-sandbox-demos/pull/242

---

#### Prerequisites

- Chrome > v120 (Open chrome://version to look up your current version)
- Enable Privacy Sandbox APIs (Open `chrome://settings/privacySandbox` to enable this setting)

We are in the process of enrolling the new endpoints. Until then, visit `chrome://flags/#privacy-sandbox-enrollment-overrides` and add the following origins to the [Privacy Sandbox Overrides textbox](https://github.com/privacysandbox/attestation/blob/main/how-to-enroll.md#5-do-i-need-to-enroll-to-test-against-local-development-environments).

Origins for the enrollment override:

```text
https://privacy-sandbox-demos-dsp-a.dev,
https://privacy-sandbox-demos-dsp-b.dev,
https://privacy-sandbox-demos-ssp-a.dev,
https://privacy-sandbox-demos-ssp-b.dev,
https://privacy-sandbox-demos-ad-server.dev
```

#### Demo

The query param that enables the video ad demo on the advertiser and publisher sites is `?auctionType=multi`.

1. Navigate to advertiser site:
   [https://privacy-sandbox-demos-shop.dev/items/1f45e?auctionType=multi](https://privacy-sandbox-demos-shop.dev/items/1f45e?auctionType=multi)
1. Navigate to publisher site: [https://privacy-sandbox-demos-news.dev/?auctionType=multi](https://privacy-sandbox-demos-news.dev/?auctionType=multi)
1. Click ‘Play video’ when the button is ready

**Comment by kevinkiklee** [comment]
Comment from @adamleslie migrated from the [original post](https://github.com/privacysandbox/privacy-sandbox-demos/pull/242#issuecomment-1934140224)

---

Ok, so lots of demand goes into the top-level auction; only one demand source comes out of the top-level auction - this would be an issue if that understanding is correct.

Summarising, moving to this model may have notable impacts to video advertising fill for Publishers - up to say 50% less fill and thus revenues.

**Detail:**
Current video advertising operations behave differently from display due to the expected functionality of VAST. This is to say that even in a scenario where you run auctions before trying to render (say through Prebid), buyers commonly run additional logic (say brand safety checks) at the point of time when the player tries to render the underlying mediafile from the VAST XML document. Today, this is not so big of an issue as most Publishers tend to populate several VAST XMLs into a playlist (say 5 per video opportunity) for the player to mediate through using VAST errors; however in a scenario where this is reduced to one there is a greater chance that it errors and causes a missed video ad opportunity.

_Furthermore,_
Considering the above point and integration with wider demand (say direct advertising campaigns which are VAST redirects or holding group multi deal IDs), it's not obvious how or when these PAAPI components from the top level auction are included into the render process (read: VAST playlist which typically is the response object to the player from the ad server) for the IMA SDK.

_In short,_
- It would be good to explore including more components from the top-level auction understanding the current behaviours of video advertising in practice (only for video)
- It would be good to understand in greater detail how this integrates into the wider advertising operations, specifically technologies such as the Google IMA SDK

If you have any further questions on the points I've made - let me know!

**Comment by kevinkiklee** [comment]
Hi @adamleslie, thanks for your detailed explanation! 

This demo is a proof-of-concept on how VASTs can work in a Protected Audience auction, and we completely agree that it's just one part of a broader discussion that should include GAM and other players in the ecosystem. We will continue to engage with various stakeholders and help them design a solution that is accepted by the ecosystem.
